File: benchmarks/primegeneration/primegeneration.cs
using System;

class PrimeGeneration
{
    // Configuration - limits for counting
    private const int LIMIT1 = 100;
    private const int LIMIT2 = 1000;
    private const int LIMIT3 = 10000;
    private const int LIMIT4 = 50000;

    // Markers matching Lamina version
    private const long HEADER_MARKER = 123456789L;
    private const long FOOTER_MARKER = 987654321L;

    static bool IsPrime(int n)
    {
        if (n <= 1) return false;
        if (n <= 3) return true;
        if (n % 2 == 0) return false;

        for (int i = 3; i * i <= n; i += 2)
        {
            if (n % i == 0) return false;
        }
        return true;
    }

    static long CountPrimes(int limit)
    {
        if (limit < 2) return 0;
        long count = 0;
        for (int i = 2; i <= limit; i++)
        {
            if (IsPrime(i)) count++;
        }
        return count;
    }

    static void Main()
    {
        // Print header marker
        Console.WriteLine(HEADER_MARKER);

        // Count and print prime counts for different limits
        Console.WriteLine(CountPrimes(LIMIT1));
        Console.WriteLine(CountPrimes(LIMIT2));
        Console.WriteLine(CountPrimes(LIMIT3));
        Console.WriteLine(CountPrimes(LIMIT4));

        // Print footer marker
        Console.WriteLine(FOOTER_MARKER);
    }
}

File: benchmarks/primegeneration/primegeneration.rs
// primegeneration.rs
// Rust implementation of prime generation benchmark

// Configuration - limits for counting
const LIMIT1: usize = 100;
const LIMIT2: usize = 1000;
const LIMIT3: usize = 10000;
const LIMIT4: usize = 50000;

// Markers matching Lamina version
const HEADER_MARKER: i64 = 123456789;
const FOOTER_MARKER: i64 = 987654321;

fn is_prime(n: usize) -> bool {
    if n <= 1 {
        return false;
    }
    if n <= 3 {
        return true;
    }
    if n % 2 == 0 {
        return false;
    }

    let mut i = 3;
    while i * i <= n {
        if n % i == 0 {
            return false;
        }
        i += 2;
    }
    true
}

fn count_primes(limit: usize) -> i64 {
    if limit < 2 {
        return 0;
    }
    let mut count = 0;
    for i in 2..=limit {
        if is_prime(i) {
            count += 1;
        }
    }
    count
}

fn main() {
    // Print header marker
    println!("{}", HEADER_MARKER);

    // Count and print prime counts for different limits
    println!("{}", count_primes(LIMIT1));
    println!("{}", count_primes(LIMIT2));
    println!("{}", count_primes(LIMIT3));
    println!("{}", count_primes(LIMIT4));

    // Print footer marker
    println!("{}", FOOTER_MARKER);
}
